src/PhpSpec/Factory/ObjectFactory.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of PhpSpec, A php toolset to drive emergent
+ * design by specification.
+ *
+ * (c) Marcello Duarte <marcello.duarte@gmail.com>
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace PhpSpec\Factory;
+
+use PhpSpec\Exception\Fracture\FactoryDoesNotReturnObjectException;
+
+class ObjectFactory
```

Comment:
Can we make this an instance rather than static calls? Because the callable isn't guaranteed pure it may have side effects